Shell script for updating shapes and api files

adam-fowler opened the pull request:
- Gets models from aws-sdk-go
- Copies them into aws-sdk-swift project
- Builds code generator
- Runs code generator
- Builds service files

Thought we could update the model files as the last build was done 9 months ago. Then thought automating this would make it a lot quicker and easier to keep these up to date.

adam-fowler commented:
Added options to 
- specify a aws-sdk-go version number 
- commit changes immediately

Yasumoto commented on scripts/update_models.sh:
```diff
@@ -0,0 +1,107 @@
+#!/bin/sh
+
+set -e
```
How about `set -eux` ? 🙏

Yasumoto commented on scripts/update_models.sh:
```diff
@@ -0,0 +1,107 @@
+
+get_aws_sdk_go()
+{
+  # clone aws-sdk-go into folder
```
Small nit, any chance you could do something like

```
DESTINATION_FOLDER="$1"
BRANCH_NAME="$2"
```

here and elsewhere to make it clear what the positional args are for? 🙏

Yasumoto commented on scripts/update_models.sh:
```diff
@@ -0,0 +1,107 @@
+    if [ -n "$LOCAL_CHANGES" ] && [ "$COMMIT_CHANGES" == 1 ]; then
+        echo "You have local changes already and you have requested to commit changes after this script has run.\nEither remove the -c option or stash your local changes."
+        usage
+        exit
```
`usage` should already call `exit 2` so this isn't necessary

Yasumoto commented on scripts/update_models.sh:
```diff
@@ -0,0 +1,107 @@
+
+commit_changes()
+{
+    COMMIT_MSG="Sync models with aws-sdk-go "$AWS_MODELS_VERSION
```
I almost always double-quote my variables in shell, so this should be:

`COMMIT_MSG="Sync models with aws-sdk-go $AWS_MODELS_VERSION"`

Yasumoto commented on scripts/update_models.sh:
```diff
@@ -0,0 +1,107 @@
+
+check_for_local_changes
+
+#create temp folder
```
I'd suggest removing these, your echo statements do a great job of describing what's going on already.

adam-fowler commented:
Updated the script with your change requests, ran shellcheck on it and fixed issues.
I'm happy for shellcheck to run on all shell scripts if you want to add it in

adam-fowler commented:
@jonnymacs @Yasumoto Are you guys happy for me to do a rebuild of all the Shape files?
